json-decode.go

```diff
@@ -12,65 +12,52 @@ import (
 //----------------------------------------
 // cdc.decodeReflectJSON
 
-// CONTRACT: rv is addressable as per https://golang.org/pkg/reflect/#Value.CanAddr
-func (cdc *Codec) decodeReflectJSON(bz []byte, info *TypeInfo, rv reflect.Value, opts FieldOptions) error {
-	// By this point, we MUST be dealing with an addressable value.
+// CONTRACT: rv.CanAddr() is true.
+func (cdc *Codec) decodeReflectJSON(bz []byte, info *TypeInfo, rv reflect.Value, opts FieldOptions) (err error) {
 	if !rv.CanAddr() {
-		panic("value is not addressable")
+		panic("rv not addressable")
 	}
-
-	// SANITY CHECK
 	if info.Type.Kind() == reflect.Interface && rv.Kind() == reflect.Ptr {
 		panic("should not happen")
 	}
 
-	// No need for diambiguation, decode as is.
+	// No need for disambiguation, decode as is.
 	if !info.Registered {
-		return cdc._decodeReflectJSON(bz, info, rv, opts)
+		err = cdc._decodeReflectJSON(bz, info, rv, opts)
+		return
 	}
 
-	// Otherwise disambiguation time
-	disfix, restBlob, err := parseDisfixAndData(bz)
+	// Otherwise, disambiguation time.
+	disfix, bz, err := decodeDisfixJSON(bz)
```

Comment:
What do you think?  I like less lines of code for this kind of operation on codecs...

Reply:
You mean shorter lines of code? Sure, LGTM.